public/scripts/popup.js

```diff
@@ -334,10 +334,21 @@ export class Popup {
             evt.preventDefault();
             evt.stopPropagation();
             await this.complete(POPUP_RESULT.CANCELLED);
-            window.removeEventListener('cancel', cancelListenerBound);
         };
-        const cancelListenerBound = cancelListener.bind(this);
-        this.dlg.addEventListener('cancel', cancelListenerBound);
+        this.dlg.addEventListener('cancel', cancelListener.bind(this));
+
+        // Don't ask me why this is needed. I don't get it. But we have to keep it.
+        // We make sure that the modal on it's own doesn't hide. Dunno why, if onClosing is triggered multiple times through the cancel event, and stopped
+        // It seems to just call 'close' on the dialog even if the 'cancel' event was prevented.
+        // Here, we just say that close should not happen if the dalog has no result.
+        const closeListener = async (evt) => {
+            if (this.result === undefined) {
```

Comment:
Could you add and use a private `closePrevented` flag instead?

Reply:
Where would you reset that flag? After the successful `onClosing` execution?

Reply:
That was the idea.

Reply:
Done.